resolwe_bio/processes/support_processors/prepare_geo.yml

```diff
@@ -0,0 +1,267 @@
+# ===========
+# Prepare GEO
+# ===========
+---
+
+- slug: prepare-geo-chipseq
+  name: Prepare GEO - ChIP-Seq
+  requirements:
+    expression-engine: jinja
+    resources:
+      cores: 1
+    executor:
+      docker:
+        image: resolwebio/base:ubuntu-17.10
+  data_name: 'Prepare GEO ({{name}} - ChIP-Seq)'
+  version: 0.0.1
+  type: data:other:geo:chipseq
+  category: analyses
+  persistence: CACHED
+  description: >
+    Prepare ChIP-Seq data for GEO upload.
+  input:
+    - name: reads
+      label: Reads
+      type: list:data:reads:fastq
+      description: >
+        List of reads objects. Fastq files will be used.
+    - name: macs14
+      label: Macs14
+      type: list:data:chipseq:macs14
+      description: >
+        List of macs14 objects. Wiggle files will be used.
+    - name: relations
+      label: Relations
+      type: list:basic:string
+      description: >
+        Sample to background mapper. Accepted format <sample name>:<background sample name>.
+    - name: name
+      label: Collection name
+      type: basic:string
+  output:
+    - name: tarball
+      label: GEO folder
+      type: basic:file
+    - name: table
+      label: Annotation table
+      type: basic:file
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+      NAME={{ name }}_ChIP-Seq
+      mkdir "${NAME}"
+
+      {% set macs_first = macs14|first %}
+
+      declare -A SAMPLE_TO_PAIRED
+      declare -A SAMPLE_TO_BACKGROUND
+
+      {% for relation in relations %}
+        SAMPLE_TO_BACKGROUND[{{ relation.split(':')[0].split('.')[0] }}]={{ relation.split(':')[1].split('.')[0] }}
+      {% endfor %}
+
+      echo -e "File_name\tSample_name\tmd5sum\tSE/PE\tRead_length" > "${NAME}".txt
+
+      {% for read in reads %}
+        SAMPLE_NAME=`basename {{ read|sample_name }} .fastq.gz`
```

Comment:
Resolwe implements the `basename` filter [here](https://github.com/genialis/resolwe/blob/master/resolwe/flow/expression_engines/jinja/filters.py#L46).

Reply:
This would become: 
```
{% set basename = read|sample_name|basename %}
{% set sample_name = basename.split('.')[0] %}
```